Objects/stringlib/unicode_format.h

```diff
@@ -1044,14 +1062,19 @@ formatteriter_next(formatteriterobject *it)
         if (conversion_str == NULL)
             goto done;
 
-        tuple = PyTuple_Pack(4, literal_str, field_name_str, format_spec_str,
-                             conversion_str);
+        PyStructSequence_InitType(&FormatterIterResultType, &formatter_iter_result_desc);
+        Py_INCREF((PyObject *) &FormatterIterResultType);
+        res = PyStructSequence_New(&FormatterIterResultType);
+
+        PyStructSequence_SET_ITEM(res, 0, literal_str); 
+        PyStructSequence_SET_ITEM(res, 1, field_name_str);
+        PyStructSequence_SET_ITEM(res, 2, format_spec_str); 
+        PyStructSequence_SET_ITEM(res, 3, conversion_str); 
+
+
     done:
-        Py_XDECREF(literal_str);
-        Py_XDECREF(field_name_str);
-        Py_XDECREF(format_spec_str);
-        Py_XDECREF(conversion_str);
-        return tuple;
+
```

Comment:
There is a leak in case of error.

Decrefs are needed only in case of error. If values are set to a named tuple, they are not needed.

Rename `done` to `error` and add `return` before it.

Reply:
Done in ff5651c.